include/deal.II/grid/grid_tools.h

```diff
@@ -1396,7 +1396,7 @@ namespace GridTools
                 // this line has children
                 cell->face(face)->child(0)->vertex(1)
                   = (cell->face(face)->vertex(0) +
-                     cell->face(face)->vertex(1)) / 2;
+                     cell->face(face)->vertex(1)) / 2.;
```

Comment:
Well, I think that's arguable. You're taking an average, so you're dividing by the number of elements you added up. That's an integer.